### PR TITLE
handle realpath in makerootfs.sh sanely

### PR DESCRIPTION
**Pull request by deitch**
We use `realpath` in `makerootfs.sh` to find the absolute path to various elements. Unfortunately, that has two shortcomings:

1. `realpath` is not available on all systems, mainly some Mac
2. even where `realpath` is available, we run it against a filename that may not yet exist. Some systems will resolve it, but others will error out.

This updates `makerootfs.sh` to handle those cases sanely. Instead of `realpath` the target _file_, we `realpath` the parent _directory_.

If `realpath` does not exist, we fall back to `cd $parent && pwd -P`, which gives the same result.

### CODE REVIEW

**Comment by eriknordmark** [comment]
> If `realpath` does not exist, we fall back to `cd $parent && pwd -P`, which gives the same result.

Why can't we always do this if it gives the same result?

File: tools/makerootfs.sh [hunk 1]
```diff
@@ -56,6 +56,37 @@ do_imagefromtar() {
...
+  dir=$(dirname "$target")
+  filename=$(basename "$target")
+  if [ ! -d "$dir" ]; then
+    echo "Error: Unable to find directory '$dir'" >&2
+    return 1
+  fi
+  if command -v realpath >/dev/null 2>&1; then
+    absolute_dir=$(realpath "$dir")
+  else
+    absolute_dir=$(cd "$(dirname "$target")"; pwd -P)
```
Comment (eriknordmark) [inline tools/makerootfs.sh:85]: You can use $dir here instead of redoing dirname.

Comment (deitch) [reply tools/makerootfs.sh]: Fixed

**Comment by deitch** [comment]
> Why can't we always do this if it gives the same result?

No idea. I guess we could. I will try it and update.

**Comment by andrewd-zededa** [comment]
Latest changes are still a working build on my machine.

**Review by eriknordmark — APPROVED** [review]
LGTM

**Comment by zedi-pramodh** [comment]
If realpath is removed then we do not need the commit. You might as well revert that commit.

commit bbce62550ec90a366c9828e6a5cc74ba71b1d7a9
Author: Pramodh Pallapothu <pramodh@zededa.com>
Date:   Tue Jun 6 12:40:58 2023 -0700

    Add coreutils package as prereq for MAC
    
    Signed-off-by: Pramodh Pallapothu <pramodh@zededa.com>

**Comment by deitch** [comment]
That commit isn't in here.

**Comment by zedi-pramodh** [comment]
I know I did that. But since you are removing realpath you might as remove the prerequisites of coreutils for the MAC.

**Comment by deitch** [comment]
No problem @zedi-pramodh ; done.

**Review by eriknordmark — APPROVED** [review]
Run eden

**Pull request merged**